### ISSUE TITLE
(batch.ManagedEc2EcsComputeEnvironment): (instance_types get a extra optimal in batch)

### ISSUE THREAD

**Issue by wttat** [aws/aws-cdk#25562]
### Describe the bug

here is my cdk code:

```
from aws_cdk import aws_batch_alpha as batch

af2_1GPU = batch.ManagedEc2EcsComputeEnvironment(
            self,"Alphafold2CE1GPU",
            vpc=vpc,
            minv_cpus=0,
            instance_types=[ec2.InstanceType.of(ec2.InstanceClass.P3, ec2.InstanceSize.XLARGE2)],
            launch_template=launch_template,
            security_groups=[
                sg,
            ],
            vpc_subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PRIVATE_WITH_EGRESS)
        )
```

### Expected Behavior

i expect that only p3.2xlarge in batch CE's Instance types

### Current Behavior

However, there are both p3.2xlarge and **optimal**

### Reproduction Steps

cdk synth

### CDK CLI Version

2.79.1

### Node.js Version

v14.16.1.

### OS

macOS 13.3.1 (a) (22E772610a)

### Language

Python

**Comment by suzhoum** [comment]
Same. I tried to use `use_optimal_instance_classes=False` to suppress the `optimal` option, but I got the following error
```
5:34:17 PM | UPDATE_FAILED        | AWS::Batch::ComputeEnvironment | agbenchtestcomputeenvironment7EDAECD2
Resource handler returned message: "null" (RequestToken: cfd95b9f-ab8b-ec3c-5c84-0b01d1a2fece, HandlerErrorCode: InternalFailure)
```

**Comment by suzhoum** [comment]
Actually removing the stack and redeploying with the `use_optimal_instance_classes=False` resolved the issue.

**Comment by peterwoodworth** [comment]
@wttat as @suzhoum mentions, the `useOptimalInstanceClasses` is set to true by default. Feel free to set this to false

@suzhoum What was the CDK diff on the update failure? Was it just removing `optimal` from the `InstanceTypes` property?

**Comment by suzhoum** [comment]
@peterwoodworth I saw another issue reported https://github.com/aws/aws-cdk/issues/21988 so followed the advice to resolve the internal failure. I was debugging another issue related to lambda not able to perform `batch:SubmitJob` after upgrading to 2.79.0 from 2.69.0. I was trying to update my code to generate the exact set of CFN that was generated in 2.69.0, that's why I ended up here. I will open another issue soon.
